azurelinuxagent/common/cgroupconfigurator.py

```diff
@@ -180,6 +185,70 @@ def disable(self):
             self._cgroups_enabled = False
             CGroupsTelemetry.reset()
 
+        @staticmethod
+        def __collect_azure_unit_telemetry():
+            azure_unit_name = None
+            azure_unit_description = None
+            try:
+                units = shellutil.run_command(['systemctl', 'list-units', 'azure*', '-all'])
+                for line in units.split('\n'):
+                    match = re.match(r'\s?(azure[^\s]*)\s?', line, re.IGNORECASE)
+                    if match is not None:
+                        azure_unit_name = match.group(1)
+                        azure_unit_description = line
+            except shellutil.CommandError as command_error:
+                message = "Failed to list systemd units: {0}".format(ustr(command_error))
+                logger.info(message)
+                add_event(op=WALAEventOperation.CGroupsInitialize, is_success=False, message=message, log_event=False)
+
+            if azure_unit_name is None:
+                return
+
+            azure_unit_location = "Unknown"
+            try:
+                azure_unit_location = SystemdCgroupsApi.get_unit_property(azure_unit_name, "Slice")
+            except Exception as exception:
+                message = "Failed to query Slice for {0}: {1}".format(azure_unit_name, ustr(exception))
+                logger.info(message)
+                add_event(op=WALAEventOperation.CGroupsInitialize, is_success=False, message=message, log_event=False)
+
+            message = "Found an Azure unit under {0}: {1}".format(azure_unit_location, azure_unit_description)
+            logger.info(message)
+            add_event(op=WALAEventOperation.CGroupsInitialize, message=message)
+
+        @staticmethod
+        def __collect_agent_unit_files_telemetry():
+            agent_unit_files = []
+            agent_service_name = get_osutil().get_service_name()
+            try:
+                fragment_path = SystemdCgroupsApi.get_unit_property(agent_service_name, "FragmentPath")
+                if fragment_path != "/lib/systemd/system/{0}.service".format(agent_service_name):
```

Comment:
As per our setup file, there are cases where the path would be `/usr/lib/systemd/system` (Eg: Redhat, Suse, etc).
Shouldn't you also add them to this check?

Reply:
right now the whole cgroups infrastructure is limited to Ubuntu. once that is stable i'll add support for other distros that also use systemd

Reply:
Ahh ok makes sense